x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_initializer.tsx

```diff
@@ -0,0 +1,155 @@
+import { MlJob } from '@elastic/elasticsearch/lib/api/types';
+import type { SingleMetricViewerServices } from '..';
+import { TimeRangeBounds } from '../../application/util/time_buckets';
+import { SeriesControls } from '../../application/timeseriesexplorer/components/series_controls';
+import {
+  APP_STATE_ACTION,
+  type TimeseriesexplorerActionType,
+} from '../../application/timeseriesexplorer/timeseriesexplorer_constants';
+
+export interface SingleMetricViewerInitializerProps {
+  bounds: TimeRangeBounds;
+  defaultTitle: string;
+  initialInput?: SingleMetricViewerServices;
+  job: MlJob;
+  onCreate: (props: {
+    panelTitle: string;
+    functionDescription: string;
+    selectedDetectorIndex: number;
+    selectedEntities: any;
+  }) => void;
+  onCancel: () => void;
+}
+
+export const SingleMetricViewerInitializer: FC<SingleMetricViewerInitializerProps> = ({
+  bounds,
+  defaultTitle,
+  initialInput,
+  job,
+  onCreate,
+  onCancel,
+}) => {
+  const [panelTitle, setPanelTitle] = useState<string>(defaultTitle);
+  const [functionDescription, setFunctionDescription] = useState<string>('');
+  const [selectedDetectorIndex, setSelectedDetectorIndex] = useState<number>(0);
+  const [selectedEntities, setSelectedEntities] = useState<any>();
+
+  const isPanelTitleValid = panelTitle.length > 0;
+
+  const handleStateUpdate = (action: TimeseriesexplorerActionType, payload: any) => {
+    switch (action) {
+      case APP_STATE_ACTION.SET_ENTITIES:
+        setSelectedEntities(payload);
+        break;
+      case APP_STATE_ACTION.SET_FUNCTION_DESCRIPTION:
+        setFunctionDescription(payload);
+        break;
+      case APP_STATE_ACTION.SET_DETECTOR_INDEX:
+        setSelectedDetectorIndex(payload);
+        break;
+      default:
+        break;
+    }
+  };
+
+  return (
+    <EuiModal
+      initialFocus="[name=panelTitle]"
+      onClose={onCancel}
+      data-test-subj={'mlSingleMetricViewerEmbeddableInitializer'}
+    >
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          <FormattedMessage
+            id="xpack.ml.SingleMetricViewerEmbeddable.setupModal.title"
+            defaultMessage="Single metric viewer configuration"
+          />
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <EuiForm>
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.ml.singleMetricViewerEmbeddable.panelTitleLabel"
+                defaultMessage="Panel title"
+              />
+            }
+            isInvalid={!isPanelTitleValid}
+          >
+            <EuiFieldText
+              data-test-subj="panelTitleInput"
+              id="panelTitle"
+              name="panelTitle"
+              value={panelTitle}
+              onChange={(e) => setPanelTitle(e.target.value)}
+              isInvalid={!isPanelTitleValid}
+            />
+          </EuiFormRow>
+
+          <SeriesControls
```

Comment:
Nit - I think we could do with an `EuiSpacer` here between the Panel title field and the controls below.

Reply:
Added in d931543fa904ca44d10ca6e19a0855dab00062d6